### PR TITLE
Add requirements and troubleshooting for ECR

### PR DESCRIPTION
**Pull request by erictune**

### CODE REVIEW

**Comment by erictune** [comment]
@therc please review docs update.

**Comment by therc** [comment]
LGTM

File: docs/user-guide/images.md [hunk 1]
```diff
@@ -69,6 +69,20 @@ images in the ECR registry.
...
+- ECR must be offered in your region
+
+Troubleshooting:
+
+- Verify all requirements above.
+- Get $REGION (e.g. `us-west-2`) credentials on your workstation. SSH into the host and run Docker manually with those creds. Does it work?
+- Verify kubelet is running with `--cloud-provider=aws`.
+- Check kubelet logs (e.g. `journalctl -t kubelet`) for log lines like:
+  - `plugins.go:56] Registering credential provider: aws-ecr-key`
+  - `provider.go:91] Refreshing cache for provider: *aws_credentials.ecrProvider`
```
Comment (johndmulhausen) [inline docs/user-guide/images.md]: I'm sure these troubleshooting steps are all accurate and great but I feel like they're quite terse. For each of these four things, can you dig up a docs link that will help them execute these steps, and be sure to specify the full command each time, not just the flags? e.g.

- Verify kubelet is running with `kubelet --cloud-provider=aws`.

File: docs/user-guide/images.md [hunk 2]
```diff
@@ -69,6 +69,20 @@ images in the ECR registry.
 The kubelet will fetch and periodically refresh ECR credentials.  It needs the
 `ecr:GetAuthorizationToken` permission to do this.
 
+Requirements:
+
+- You must be using kubelet version `v1.2.0` or newer.  (e.g. run `/usr/bin/kubelet --version=true`).
```
Comment (johndmulhausen) [inline docs/user-guide/images.md]: How about:

- You must be using kubelet version `v1.2.0` or newer. Run this to check:

  ```shell
  kubelete --version=true
  ```

Comment (therc) [inline docs/user-guide/images.md]: They might also be on CoreOS, which comes with an ancient kubelet in /usr/bin (I put my own binary elsewhere). In that case they would have to double check in systemd, upstart or initd configuration files which binary is actually spawned. I have a PR to just log the version at startup.

**Comment by kelseyhightower** [comment]
@erictune Can you rebase this so I can merge it? It's not perfect but it gets the job done.

**Comment by erictune** [comment]
@kelseyhightower done.

**Pull request merged**